students/synowiec_krzysztof/lesson_12_debugging/validated_user_base.py

```diff
@@ -0,0 +1,121 @@
+import re
+import os
+import argparse
+import sys
+import csv
+import logging
+logging.basicConfig(format='%(asctime)s - %(levelname)s- %(message)s')
+
+EMAIL_K = 'email'
+PASS_K = 'password'
+PHONE_K = 'phone'
+POST_K = 'postal code'
+FILEDNAMES = [EMAIL_K, PASS_K, PHONE_K, POST_K]
+
+
+def main():
+    file_name = 'data.csv'
+    user_map = {}
+
+    if os.path.exists(file_name):
+        for row in csv.DictReader(open(file_name, 'r')):
+            user_map[row[EMAIL_K]] = (row[PASS_K], row[PHONE_K], row[POST_K])
+    logging.debug('Data file contains {} records'.format(len(user_map)))
+
+    e, p, m, c, v = check_args(sys.argv[1:])
+    log_level = get_log_level(v)
+    logging.getLogger().setLevel(log_level)
+    combined_validator(e, p, m, c)
+    logging.info('{} {} {} {} accepted'.format(e, p, m, c))
+    if e in user_map:
+        logging.info('Email {} is in file. Data will be updated'.format(e))
+    user_map[e] = (p, m, c)
+
+    writer = csv.DictWriter(open(file_name, 'w'), FILEDNAMES)
+    writer.writeheader()
+    for email in user_map:
+        passw, phone, postal = user_map.get(email)
+        row = {EMAIL_K: email, PASS_K: passw, PHONE_K: phone, POST_K: postal}
+        writer.writerow(row)
```

Comment:
After few runs my data.csv file looks like:

```
email,password,phone,postal code

abc@gmail.com,q,123123123,88-888

abc1@gmail.com,q,123123123,88-888

abc2@gmail.com,q,123123123,88-888
```
There definitely should not be blank lines between entries. It's only on my computer ? Please check that out.